### PR TITLE
Add paz.transformers toolkit and port Whisper decoding onto it

### PR DESCRIPTION
**Pull request by oarriaga**
## What

Introduces `paz/models/transformers` — surfaced as **`paz.transformers`** — a
small, functional toolkit of reusable sequence-model primitives, and ports the
Whisper example (`examples/speech_to_text`) onto it. First step toward sharing
transformer machinery across Whisper and Gemma4 (Gemma4 follows in a later PR).

### `paz.transformers` (new)

Module names the object/process; functions are bare verbs; params are explicit
(no `*Args` bundles). Each with a co-located `*_test.py`.

- `cache.build` / `cache.update` — KV cache `(batch, layers, 2, len, heads, dim)`
- `search.build` — jitted autoregressive decode loop; selectors `search.greedy`
  and `search.build_sampler(temperature, top_k, top_p)`
- `logits.apply_temperature` / `apply_top_k` / `apply_top_p`
- `mask.causal`
- `feedforward.gelu` (vanilla) / `feedforward.glu` (gated GeGLU)

### Whisper refactor (behavior-preserving)

`decoding.py` and `layers/{attention,decoder,encoder}.py` now use the toolkit.
QKV projection, cross-attention, learned positions and the warmup/prefill stay
model-side. Greedy decoding is unchanged — generated token ids are **identical**
before and after on `whisper_tiny_en` and `whisper_base_en` (on `harvard.wav`).
The demo gains optional `--temperature/--top_k/--top_p` sampling (greedy stays
the default) via `paz.transformers.search.build_sampler`.

### Incidental fix

`model.py` imported `ReversibleEmbedding` from `keras.layers`, which is not
present on the current keras; corrected to `keras_hub.layers`, matching
`examples/gemma4`, so the example loads.

## Validation

- Greedy token ids identical pre/post refactor (`whisper_tiny_en`,
  `whisper_base_en`).
- Sampling: `top_k=1` reduces to greedy; reproducible under a fixed key.
- `pytest paz/models/transformers/` → 13 passed.
- `pytest examples/speech_to_text/push_to_talk_test.py` → 10 passed.
  (Run with `KERAS_BACKEND=jax`, as for the rest of the JAX-first repo.)

## Notes

The branch is based on an earlier `paz-jax`; `paz/__init__.py` and
`examples/speech_to_text/` are unchanged on `paz-jax` since, so the diff is
conflict-free. Happy to rebase if preferred.

### CODE REVIEW

File: examples/speech_to_text/decoding.py [hunk 1]
```diff
@@ -2,97 +2,75 @@
...
 
+import paz
 from examples.speech_to_text.tokenizer import find_special_token_id
 
 PROMPT_TOKENS = ["<|startoftranscript|>", "<|transcribe|>", "<|notimestamps|>"]
 
 
-def kv_decode(decoder, cache_shape, cross_model, encoder_output, stop_id):
+def kv_decode(decoder, cache_shape, cross_model, encoder_output, stop_id,
+              key=None):
```
Comment (oarriaga) [inline examples/speech_to_text/decoding.py]: We should always get a key.

File: examples/speech_to_text/demo.py [hunk 2]
```diff
@@ -36,10 +39,10 @@
 MAX_AUDIO_SECONDS = 30
 
 
-def transcribe(wav_path, models, max_tokens=64):
+def transcribe(wav_path, models, max_tokens=64, select=None, key=None):
```
Comment (oarriaga) [inline examples/speech_to_text/demo.py]: We should always pass a valid key not None. Also, key should be usually the first argument of a function.

File: examples/speech_to_text/decoding.py (hunk 1 above)
Comment (oarriaga) [reply examples/speech_to_text/decoding.py]: Done in 40130157 — `kv_decode` now requires `key` as its first argument; dropped the `key=None` default and the `PRNGKey(0)` fallback. Same for the jitted `decode` and `search.build`'s `run`.

File: examples/speech_to_text/demo.py (hunk 2 above)
Comment (oarriaga) [reply examples/speech_to_text/demo.py]: Done in 40130157 — the demo always builds `key = jax.random.PRNGKey(args.seed)` (greedy ignores it), and `key` is now the first argument across the transcribe chain. No more `None` keys.
